rust/src/transformations/make_stable_lazyframe/filter/test.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    domains::{AtomDomain, LazyFrameDomain, Margin, OptionDomain, SeriesDomain},
+    metrics::SymmetricDistance,
+    transformations::make_stable_lazyframe,
+};
+
+use super::*;
+
+#[test]
+fn test_filter() -> Fallible<()> {
+    let lf = df!("chunk_2_null" => [[Some(1i64); 500], [None; 500]].concat())?.lazy();
+
+    let lf_domain = LazyFrameDomain::new(vec![SeriesDomain::new(
+        "chunk_2_null",
+        OptionDomain::new(AtomDomain::<i64>::default()),
+    )])?
+    .with_margin(&["chunk_2_null"], Margin::new().with_public_keys())?;
+
+    let t_filter = make_stable_lazyframe(
+        lf_domain.clone(),
+        SymmetricDistance,
+        lf.clone().filter(col("chunk_2_null").is_not_null()),
+    )?;
+
+    let actual = t_filter.invoke(&lf)?.collect()?;
+    assert_eq!(actual, df!("chunk_2_null" => [Some(1); 500])?);
+
+    assert!(t_filter
+        .output_domain
+        .margins
+        .values()
+        .all(|m| { m.public_info.is_none() }));
+
+    Ok(())
+}
+
+#[test]
+fn test_fail_filter() -> Fallible<()> {
```

Comment:
```suggestion
fn test_filter_fail_with_non_bool_predicate() -> Fallible<()> {
```
A little more info about what is being exercised.